spec/advance_enumerable_spec.rb

```diff
@@ -0,0 +1,163 @@
+      expect { arr.my_select(1) }.to raise_error(ArgumentError)
+    end
+    it 'returns an error if something else is given instead of an array' do
+      expect { 1.my_select }.to raise_error(NoMethodError)
+    end
+  end
+
+  describe '#my_all?' do
+    it 'returns true if all the items of array satisfies the condition' do
+      expect(arr.my_all? { |x| x < 1 }).to eql(arr.all? { |x| x < 1 })
+    end
+    it 'returns true if pattern is matched or array has all elements same as argument' do
+      expect(arr.my_all?(/d/)).to eql(arr.all?(/d/))
+    end
+    it 'returns true if all the elements are true if we dont pass block and argument' do
+      expect(arr.my_all?).to be true
+    end
+    it 'returns true if an empty array is given' do
+      expect([].my_all?).to be true
+    end
+    it 'returns an error if something else is given instead of an array' do
+      expect { 1.my_all? }.to raise_error(NoMethodError)
+    end
+  end
+
+  describe '#my_none?' do
+    it 'returns true if none of the items of array satisfies the condition' do
+      expect(arr.my_none? { |x| x < 1 }).to eql(arr.none? { |x| x < 1 })
+    end
+    it 'returns true if none of the items match the pattern passed as argument' do
+      expect(arr.my_none?(/s/)).to eql(arr.none?(/s/))
+    end
+    it 'returns true if none of the elements are true if we dont pass block and argument' do
+      expect(arr_nil.my_none?).to be true
+    end
+    it 'returns true if an empty array is given' do
+      expect([].my_none?).to be true
+    end
+    it 'returns an error if something else is given instead of an array' do
+      expect { 1.my_none? }.to raise_error(NoMethodError)
+    end
+  end
+
+  describe '#my_any?' do
+    it 'returns true if any of the items of array satisfies the condition' do
+      expect(arr.my_any? { |x| x < 1 }).to eql(arr.any? { |x| x < 1 })
+    end
+    it 'returns true if any of the items match the pattern passed as argument' do
+      expect(arr.my_any?(/d/)).to eql(arr.any?(/d/))
+    end
+    it 'returns true if any of the elements are true if we dont pass block and argument' do
+      expect(arr.my_any?).to be true
+    end
+    it 'returns flase if an empty array is given' do
+      expect([].my_any?).to be false
+    end
+    it 'returns an error if something else is given instead of an array' do
+      expect { 1.my_any? }.to raise_error(NoMethodError)
+    end
+  end
+
+  describe '#my_count' do
+    it 'counts the items of an array' do
+      expect(arr.my_count).to eql(arr.count)
+    end
+    it 'returns the amount of times an item is in an array if an argument is passed' do
+      expect(arr.my_count(2)).to eql(1)
+    end
+    it 'returns the amount of times a condition is satisfied by each element of array' do
+      expect(arr.my_count(&:even?)).to be 1
+    end
+    it 'returns an error if something else is given instead of an array' do
+      expect { 1.my_all? }.to raise_error(NoMethodError)
```

Comment:
You should change the title of your test since you can actually use the method on other enumerables beside arrays.